kernel/core/utils/stringtools.hpp

```diff
@@ -130,6 +131,241 @@ enum TimeDateFormat
+    int checkNumFormat(std::string key)
+    {
+        // TODO what about encapsulation ? this should be private then
+        auto elem = num_format_lookup.find(key);
+
+        if(elem != num_format_lookup.end())
+            return elem->second;
+
+        return 0;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function pushes digits # inbetween seperators
+    ///
+    /// \param idx int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void pushInbetween(int idx)
+    {
+        int digs_inbetween = idx - m_last_idx;
+        if(digs_inbetween > 0)
+        {
+            if(digs_inbetween < 3)
+                m_tape.push_back('<');
+            else if(digs_inbetween > 3)
+                m_tape.push_back('>');
+            else
+                m_tape.push_back('=');
+        }
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function starts parse of a number
+    ///
+    /// \param idx int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void startParseNumber(int idx)
+    {
+        m_last_idx = idx-1; //todo wie handeln wir 1ste und letzte, da diese nicht seperatoren sind
+        m_tape = "";
+        m_curr_format = 0;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called for every found seperator and adds to the tape
+    ///
+    /// \param sep char
+    /// \param idx int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void addSeperator(char sep, int idx)
+    {
+        pushInbetween(idx-1);
+        m_tape.push_back(sep);
+        m_last_idx = idx;
+
+        // max 3 symbols at ones in tape
+        while(m_tape.length() > 3)
+            m_tape.erase(m_tape.begin());
+
+        m_curr_format |= checkNumFormat(m_tape);
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called when last char of a number is reached during parsing
+    ///
+    /// \param idx int
+    /// \return int
+    ///
+    /////////////////////////////////////////////////
+    int endParseNumber(int idx)
+    {
+        // we only want to check for the last 2/1
+        m_tape = m_tape.back();
+
+        pushInbetween(idx); //since normally the function is called when seperator appears
+        m_curr_format |= checkNumFormat(m_tape);
+
+        m_last_idx = idx;
+        m_tape = "";
+        return m_curr_format;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called to make a vote with the given number Type
+    ///
+    /// \param numType int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void vote(int numType)
+    {
+        if((numType & NUM_DECIMAL_EU && numType & NUM_K_US) ||
+                (numType & NUM_DECIMAL_US && numType & NUM_K_EU))

```

Comment:
```suggestion
        if (numType & (NUM_DECIMAL_EU | NUM_K_US)
            || numType & (NUM_DECIMAL_US | NUM_K_EU))
```
Might be easier to read

Reply:
Forget that, that's not the same thing as the original code